src/Arbitrary.hs

```diff
@@ -385,7 +385,7 @@ instance Arbitrary Timestamp where
     arbitrary = Timestamp <$> arb
 
 instance GenArbitrary r => GenArbitrary (Action r) where
-    genArbitrary = liftIO $ generate arbitrary
+    genGen = liftIO . generate
```

Comment:
Actually with DefaultSignatures this is quite decent. Here is what we should do for most those IO actions:

```
class Foo m where
  foo :: A -> B -> m C
  default foo :: MonadIO m => A -> B -> m C
  foo a b = liftIO $ Orig.foo a b
```